### PR TITLE
Allow to add a collation to a query

### LINKED ISSUE #9081: Panache MongoDB - Use collation in find
**Description**
It is not possible to define a collation when executing a find with the MongoDB PanacheQuery. We want to implement filtering and sorting and apply a collation to be able to enable e.g. find case insensitive.

**Implementation ideas**
In the Zulip chat Loïc MATHIEU suggested to provide something like:

```
Person.find("description", "full text keywords").withCollation("{// your collection document}").list();
```

This would serve our needs.

### PR DESCRIPTION
**Pull request by loicmathieu**
Fixes #9081

### CODE REVIEW

File: extensions/panache/mongodb-panache/runtime/src/main/java/io/quarkus/mongodb/panache/runtime/PanacheQueryImpl.java [hunk 1]
```diff
@@ -167,6 +178,9 @@ public long count() {
         if (this.projections != null) {
             find.projection(projections);
         }
+        if (this.collation != null) {
```
Comment (FroMage) [inline extensions/panache/mongodb-panache/runtime/src/main/java/io/quarkus/mongodb/panache/runtime/PanacheQueryImpl.java:181]: Is there a good reason the reactive/non-reactive queries have code that is different here? reactive has 2 `buildOptions` methods while here it's done in `list`. Is is normal or a sign that one of the two has a refactoring missing?

**Review by FroMage — APPROVED** [review]
LGTM if my question does not point to an issue :)

File: extensions/panache/mongodb-panache/runtime/src/main/java/io/quarkus/mongodb/panache/runtime/PanacheQueryImpl.java (hunk 1 above)
Comment (loicmathieu) [reply extensions/panache/mongodb-panache/runtime/src/main/java/io/quarkus/mongodb/panache/runtime/PanacheQueryImpl.java:181]: Only the reactive API have the ability to pass an option to a Query.
The imperative must mutate the query before iteratinge the cursor.

**Comment by geoand** [comment]
So are we good with this one?

**Comment by loicmathieu** [comment]
@FroMage is it OK for you? Can we merge ?

**Comment by FroMage** [comment]
Yup.

**Pull request merged**